### PR TITLE
A proposal of how usage examples could look like.

### PR DESCRIPTION
**Pull request by yundt**
A proposition of how usage examples nested in the orb structure could look like.

An array of `examples` being maps allows for adding more data later (e.g. a `featured` flag for generating documentation).

This still doesn't solve the problem of how to validate example usages against the full schema or expand the orb itself to compare with the expected result, but I believe it's a flexible starting point.

Comments are welcome!

### CODE REVIEW

File: docs/usage-examples.md [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+        version: 2
+        workflow:
+          jobs:
+          - build
+```
+
+When validating orb's source, the `result` map will be automatically compared with the actual output of compiling `usage`, and an error will be raised in case of a mismatch.
+
+## Example usage syntax
+An example usage map can have the following keys:
```
Comment (eric-hu) [inline docs/usage-examples.md]: Since the keys below talk about what's optional and required, this section should make clear that the `examples` parent key is optional and these apply only if a user tries to define an example.

File: docs/usage-examples.md [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+          - run:
+              command: echo Hello Anna
+      workflows:
+        version: 2
+        workflow:
+          jobs:
+          - build
+```
+
+When validating orb's source, the `result` map will be automatically compared with the actual output of compiling `usage`, and an error will be raised in case of a mismatch.
```
Comment (eric-hu) [inline docs/usage-examples.md]: Nit: The "result" comparison is possibly too large in scope for our initial work. We should rephrase this to say "in the future we may add automated example compilation and compare the output with `result`".

Comment (yundt) [reply docs/usage-examples.md]: I would go even further and remove `result` from the schema completely, initially - if we don't, we risk "breaking" already published orbs in the future when we roll this check out.

@ndintenfass, do you have objections to that?

File: docs/usage-examples.md (hunk 1 above)
Comment (yundt) [reply docs/usage-examples.md]: 👍 adding that.

File: docs/usage-examples.md (hunk 2 above)
Comment (yundt) [reply docs/usage-examples.md]: As `result` support is already deployed and it's trivial to get actual compilation output with the CLI, I'm only amending the docs to warn about a possibility of automated checks in the future.

Comment (eric-hu) [reply docs/usage-examples.md]: That sounds like a good middle ground. Thanks for updating this to match the merged work!
